api/src/gotEnzymes/crossReferencesMapping.js

```diff
@@ -51,4 +51,12 @@ export default {
     dbPrefix: 'reactome',
     compoundSuffix: '',
   },
+  ncbigene: {
+    db: 'NCBI Gene',
+    dbPrefix: 'ncbigene',
+  },
+  uniprot: {
+    db: 'UniProtKB',
+    dbPrefix: 'uniprot',
+  },
```

Comment:
It seems like the `/identifier/...` pages are not working correctly for these new reference types. For example:
- `/identifier/NCBI%20Gene/BAA98448?referenceType=gene`
- `/identifier/UniProtKB/Q9Z8U6?referenceType=gene`

Edit: Actually this seems to be for all the new reference types, for example `/identifier/SABIO-RK/321?referenceType=domain` does not work either. This is because we have the following lines in the `idInModels.js` file.

```javascript
    const VALID_REFERENCE_TYPES = ['compound', 'reaction'];
    if (!VALID_REFERENCE_TYPES.includes(referenceType)) {
      throw e;
    }
```

Reply:
Nice catch. I'm not sure why these are needed - is it perhaps because of the Neo4j query that follows it?

Reply:
We are using the `referenceType` to construct the outgoing URL to the external sites. These need to be manually defined in the `crossReferencesMapping.js` file. Currently we have `compoundSuffix` and `reactionSuffix` so I think we need to add all of the other types too.

```javascript
    const dbSuffix = dbMapping[`${referenceType}Suffix`];
...
        url: `https://identifiers.org/${dbPrefix}${dbSuffix}:${externalId}`,
```

Reply:
But that would lead to a bad URL to `identifiers.org`, which I don't think is the problem here.

Reply:
Can you please specify what the problem you are referring to is? The problem I want to solve is that these pages are not loading because we throw an error as mentioned in the initial comment because we have provided an invalid reference type by our definition of `VALID_REFERENCE_TYPES`.

Reply:
@mihai-sysbio: Me and @inghylt playde around with this. We made these changes https://github.com/MetabolicAtlas/MetabolicAtlas/commit/263550d2ad6567f16e095ce074f2ced90b7ae8b3
which fixes the problem mentioned above and in my comment below.
For ncbi genes, we noted that the identifiers often seem to refer to protein structures. For example, zwf links to BAA98582 which can be found at  https://identifiers.org/ncbiprotein:BAA98582 but not at  https://identifiers.org/ncbigene:BAA98582 . 

Is this expected? As far as we have seen, this change does not break anything.

Which databases are expected to have genes? We only find examples of NCBI, Uniprot and  MetaNetX

Reply:
@MalinAhlberg and @inghylt Those gene related changes look good I think! For the other new types, I think we also need to take care of `compound`. `domain` and `organism` don't have any cross references data so they don't need to be addressed now.

The other changes in the linked PR, removing the white space and resetting cross references, have been addressed in other commits in this current PR.

Reply:
The first time around I had added these as `ncbigene` but then I was corrected that the mapping provided by the researchers is from GotEnzymes Gene to NCBI Protein. Even in the links provided above, only the link to the protein works. I have already asked the researchers to improve the Gene cross-references, and they did. So there's little to improve in this regard.

Thanks for fixing the code checking for the `VALID_REFERENCE_TYPES`. I'm not sure why the `if` is needed there though, since I don't think this is user-reachable, so I would remove it.
```
    const VALID_REFERENCE_TYPES = ['compound', 'reaction'];
    if (!VALID_REFERENCE_TYPES.includes(referenceType)) {
      throw e;
    }
 ```

Reply:
Commit c6d86b6 fixes the cross references for genes. Compounds already work.

Reply:
Nice! I was confused about `compound` before as I thought it was a new type we added.

About the `if` block, if we remove it then it would cause `dbSuffix` to be `undefined`, which would result in a success (`200`) response with an `externalDb` object that has an invalid `url`. Technically this should not be a problem for users of our website but imo this is good practice for API design to return an error response when appropriate instead of a success response with erroneous data. Considering that we would like to expose API endpoints for `GotEnzymes`, this is especially important to get right.

Reply:
Maybe there's something I'm missing.

Given that the `dbSuffix` is simply appended (`${dbPrefix}${dbSuffix}`), the `url` to Identifiers.org could be fixed by changing the `dbPrefix` to include the suffix. Also, in [`genes.js`](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1062/files#diff-097c60247e7d6d4d84a07c1519e1a14279ef626455f51adadbc30607e976a5c8) the suffix is doesn't seem to be used.

Reply:
I would fetch the `reaction` or `metabolite` (or `gene`) from the tag of the node that is associated with the external ID in Neo4j. And if there is no component in the Neo4j database linked to that external identifier, then the information that needs to be shown is much shorter, so for these cases I would skip mentioning the `referenceType` entirely.
By the way, this refactoring sounds like a new issue to my ears.

Reply:
For this conversation, I think we can focus on the components that do no exist in the neo4j database because that's where [the relevant code](https://github.com/MetabolicAtlas/MetabolicAtlas/blob/main/api/src/neo4j/queries/idInModels.js#L50-L76) is.

The question is: Given that a user is looking for component `X` in database `Y`, how do you construct the `identifiers.org` URL? More specifically, how do the following get mapped if we want to skip mentioning of the `referenceType` entirely?

| frontend URL | identifiers.org URL |
| --- | --- |
| `/identifier/KEGG/C00003` | `https://identifiers.org/kegg.compound:C00003` |
| `/identifier/MetaNetX/MNXM22270` | `https://identifiers.org/metanetx.chemical:MNXM22270` |
| `/identifier/ChEBI/18050` | `https://identifiers.org/CHEBI:18050` |

Reply:
Had a chat with @e0 which really helped. The key message is that some databases such as `KEGG` or `MetaNetX` are namespaces which require specifying somehow if it's a `compound`, because there are others also such as `reaction` or `gene`. This is not needed for other databases such as `ChEBI`, and it's not needed for Neo4j queries since there we know what the cross-reference is mapped to in the model.

This is worth clearing up in a separate issue.